usl_pipeline/study_area_uploader/tests/transformers/test_study_area_transformers.py

```diff
@@ -98,3 +123,185 @@ def test_transform_shape_file_with_polygon_masks():
         assert soil_classes[0][1] == 8
         assert soil_classes[1][0].equals(bbox_polygon(1, 0, 3, 2))
         assert soil_classes[1][1] == 9
+
+
+def test_prepare_and_upload_study_area_files_no_boundaries_buildings_only():
+    with tempfile.TemporaryDirectory() as temp_dir:
+        work_dir = pathlib.Path(temp_dir)
+        elevation_input_file_path = work_dir / "input_elevation.tif"
+        prepare_elevation_data(
+            elevation_input_file_path,
+            [
+                [0.0, 1.0, 2.0, 3.0, 4.0],
+                [5.0, 6.0, 7.0, 8.0, 9.0],
+                [10.0, 11.0, 12.0, 13.0, 14.0],
+                [15.0, 16.0, 17.0, 18.0, 19.0],
+                [20.0, 21.0, 22.0, 23.0, 24.0],
+            ],
+        )
+        buildings_input_file_path = work_dir / "buildings.shp"
+        prepare_shape_file(
+            buildings_input_file_path,
+            [(bbox_polygon(2, 1, 4, 3), 1)],
+        )
+        study_area_bucket = mock.MagicMock()
+        buildings_cloud_storage_buffer = StringIO()
+        buildings_cloud_storage_buffer.close = lambda: None  # Want to check content
+        buildings_blob = study_area_bucket.blob.return_value
+        buildings_blob.open.return_value = buildings_cloud_storage_buffer
```

Comment:
You can avoid creating that `buildings_blob` object and do
```python
        study_area_bucket.blob.return_value.open.return_value = (
            buildings_cloud_storage_buffer
        )
```
you can also do
```
        study_area_bucket.blob().open.return_value = buildings_cloud_storage_buffer
        study_area_bucket.reset_mock()
```
note that you need the `rest_mock()` call here to remove that `blob()` call from the mock calls. Using `blob.return_value` rather than `blob()` above avoids this.

Reply:
Done